sshconnect2.c

```diff
@@ -217,7 +217,7 @@ ssh_kex2(struct ssh *ssh, char *host, struct sockaddr *hostaddr, u_short port,
     const struct ssh_conn_info *cinfo)
 {
 	char *myproposal[PROPOSAL_MAX] = { KEX_CLIENT };
-	char *s, *all_key;
+	char *s, *all_key, *hostkeyalgs;
```

Comment:
plz initialize hostkeyalgs to NULL so that free'ing it is always safe (usage below looks fine, but this might prevent problems in the future).